test/e2e/initcontainers/initcontainers_test.go

```diff
@@ -90,4 +94,25 @@ func TestInitContainers(t *testing.T) {
 	); err != nil {
 		t.Fatalf("The endpoint %s for Route %s didn't serve the expected text %q: %v", url, names.Route, test.EmptyDirText, err)
 	}
+
+	revisionName, err := e2e.RevisionFromConfiguration(clients, names.Config)
+	if err != nil {
+		t.Fatalf("Failed to get revision from configuration %s: %v", names.Config, err)
+	}
+
+	if err := v1test.CheckRevisionState(clients.ServingClient, revisionName, func(r *v1.Revision) (bool, error) {
+		if len(r.Status.InitContainerStatuses) != 1 {
+			return true, errors.New("init image digest resolution failed")
+		}
+		initContainerStatus := r.Status.InitContainerStatuses[0]
+		if validDigest, err := shared.ValidateImageDigest(t, names.Image, initContainerStatus.ImageDigest); !validDigest {
```

Comment:
This should be moved under `conformance` in the future (same as the sidecar containers status validation).